src/cargo/ops/cargo_package.rs

```diff
@@ -202,19 +202,34 @@ pub fn package(ws: &Workspace<'_>, opts: &PackageOpts<'_>) -> CargoResult<Vec<Fi
         // below, and will be validated during the verification step.
     }
 
+    let deps = local_deps(pkgs.iter().map(|(p, f)| ((*p).clone(), f.clone())));
     let just_pkgs: Vec<_> = pkgs.iter().map(|p| p.0).collect();
-    let publish_reg = get_registry(ws.gctx(), &just_pkgs, opts.reg_or_index.clone())?;
-    debug!("packaging for registry {publish_reg}");
+
+    let sid = match get_registry(ws.gctx(), &just_pkgs, opts.reg_or_index.clone()) {
+        Ok(sid) => {
+            debug!("packaging for registry {}", sid);
+            Some(sid)
+        }
+        Err(e) => {
+            if deps.has_no_dependencies() && opts.reg_or_index.is_none() {
+                // The publish registry doesn't matter unless there are local dependencies,
+                // so ignore any errors if we don't need it. If they explicitly passed a registry
+                // on the CLI, we check it no matter what.
+                None
```

Comment:
I feel uncomfortable taking any error a function may return and ignoring it based on certain criteria, assuming its a very specific case.  Someone could refactor that function, changing error cases and not realize that it doesn't propagate through.

Reply:
Good point, what about this new version? It just avoids calling the function altogether instead of ignoring the error.